### PR TITLE
Remove runtime guard for H3 Header normalisation

### LINKED ISSUE #30428: envoy_reloadable_features_use_http3_header_normalisation deprecation
Your change #26318 (Add HTTP/3 header normalisation support for extended connect) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by asingh-g**
Commit Message: Remove header guards for H3 header normalisation that were introduced during the CONNECT UDP implementation
Additional Description: It has been well over 6 months since this was introduced so this cleanup is overdue
Risk Level: Low
Testing:
Docs Changes:
Release Notes: Removed runtime guard envoy_reloadable_features_use_http3_header_normalisation
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fixes #30428
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by DavidSchinazi — APPROVED** [review]
Thanks for doing this! I would add some details to the description to point to the PR that added this flag and confirm that's it's been in there for more than 6 months

**Comment by yanavlasov** [comment]
Please add release notes as in the #35385

/wait

**Comment by asingh-g** [comment]
Updated release notes

**Comment by yanavlasov** [comment]
I see main merge but no updates to release notes. Maybe you forgot to push the change?

/wait

File: changelogs/current.yaml [hunk 1]
```diff
@@ -71,6 +71,11 @@ bug_fixes:
     Add runtime guard for timeout error code 504 Gateway Timeout that is returned to downstream. If runtime flag
     ``envoy.reloadable_features.ext_proc_timeout_error`` is set to false, old error code 500 Internal Server Error will be returned.
 
+removed_config_or_runtime:
+- area: http
```
Comment (yanavlasov) [inline changelogs/current.yaml:75]: Please move under the existing `removed_config_or_runtime` section.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
Please also fix DCO.

/wait

**Comment by asingh-g** [comment]
Huh something went wrong with that rebase, need to close this

**Pull request closed without merging**